src/library/modules/BattlePrediction/rank/Battle.js

```diff
@@ -0,0 +1,73 @@
+(function () {
+  // Based on: https://github.com/andanteyk/ElectronicObserver/blob/master/ElectronicObserver/Other/Information/kcmemo.md#%E6%88%A6%E9%97%98%E5%8B%9D%E5%88%A9%E5%88%A4%E5%AE%9A
+  const predict = (initialFleets, resultFleets) => {
+    const {
+      getSunkCount,
+      getShipCount,
+      getDamageGauge,
+      battle: { isPlayerNoDamage, isEnemyFlagshipSunk, isPlayerFlagshipTaiha },
+    } = KC3BattlePrediction.rank;
+
+    const sunk = getSunkCount(resultFleets);
+    const ships = getShipCount(resultFleets);
+    if (sunk.player === 0) {
+      if (sunk.enemy === ships.enemy) {
+        return isPlayerNoDamage(initialFleets, resultFleets) ? 'SS' : 'S';
+      }
+      if (ships.enemy > 1 && sunk.enemy >= Math.floor(ships.enemy * (2 / 3))) {
```

Comment:
wondering where is `2/3` from? EO's memo using `0.7` and they do not full equal for combined fleet:
```
> [1,2,3,4,5,6,7,8,9,10,11,12].map(v => Math.floor(v*(2/3)))
[0, 1, 2, 2, 3, 4, 4, 5, 6, 6, 7, 8]
> [1,2,3,4,5,6,7,8,9,10,11,12].map(v => Math.floor(v*0.7))
[0, 1, 2, 2, 3, 4, 4, 5, 6, 7, 7, 8]
```

Reply:
I don't remember where exactly I pulled it from, I was looking at a bunch of sources. I'll change it to match EO.

Honestly, I'm not particularly happy with how that function turned out, I might go back and rewrite it later.